fw/http_msg.c

```diff
@@ -606,7 +606,7 @@ tfw_http_msg_hdr_close(TfwHttpMsg *hm)
 	/* Allocate some more room if not enough to store the header. */
 	if (unlikely(id == ht->size)) {
 		if (tfw_http_msg_grow_hdr_tbl(hm))
-			return TFW_BLOCK;
+			return TFW_BAD;
```

Comment:
Maybe it's not subject of current PR, but error handling looks bad for me. 

We have `T_BAD/TFW_BAD` that used widely, but there is no sense. For instance: `tfw_http_msg_hdr_close` has few return paths and all of them return `TFW_PASS` or `TFW_BAD`. It looks strange, I can turn return type into boolean and won't lose anything. It's not an error handling, we must return error codes that corresponding to type of an error. If hdr table not grown due to lack of memory we should return `ENOMEM` and so on. Even in this function we call `tfw_h2_hdr_map` that returns two different error codes, but we ignore them.

Also as I believe we must  not use `TFW_*` or `T_*` return codes in functions that doesn't make a decisions about block/close/reply. If function just add headers to req and use one of this error codes that makes illusion that this function can decide whether the connection will be closed or not.

And another one, gfsm's return codes `TFW_*` used in many places even not related to gfsm, it's not good decision. We must use appropriate error codes for each module. In opinion `tfw_http_req_parse_*` should return `T_*` error codes. In other word functions that calls  `tfw_http_msg_process_generic` and `tfw_http_msg_process` expects the error codes that represents "fate" of the connection, but rest of functions must return more specific error codes.

Reply:
@krizhanovsky Please look at message above. What do you think?